File: java-module/src/test/java/com/example/BasicUsage.java
package com.example;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import org.junit.Test;

import java.io.IOException;

public class BasicUsage {
    @Test
    public void jsonRequestTest() {
        OkHttpClient client = new OkHttpClient();

        Request request = new Request.Builder()
                .url("https://naver.com")
                .build();

        try (Response response = client.newCall(request).execute()) {
            assert response.code() == 200;
            assert response.body().bytes().length > 0;
        } catch (IOException e) {
            // do nothing
        }
    }
}

File: kotlin-module/src/test/kotlin/com/example/BasicUsage.kt
package com.example

import okhttp3.OkHttpClient
import okhttp3.Request
import org.junit.Test

class BasicUsage {
    @Test
    fun sayHelloTest() {
        val client = OkHttpClient()

        val request = Request.Builder()
            .url("https://naver.com")
            .build()

        val response = client.newCall(request).execute()
        response.use {
            assert(it.code == 200)
            assert(it.body!!.bytes().isNotEmpty())
        }
    }
}